.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,159 @@
+    runs-on: 'ubuntu-latest'
+    needs: [quay-image, clair-image, builder-image, qemu-builder-image]
+    env:
+      BUNDLE: quay.io/projectquay/quay-operator-bundle
+      INDEX: quay.io/projectquay/quay-operator-index
+      TAG: 3.5
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+        # runs on main branch to use newer OLM bundle files.
+
+      - name: Login to Quay.io
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USER }}
+          password: ${{ secrets.QUAY_TOKEN }}
+
+      - name: Update CSV Image Versions
+        uses: mikefarah/yq@master
+        with:
+          cmd: |
+            yq eval -i '
+              .metadata.name = "quay-operator-test" |
+              .metadata.annotations.quay-version = strenv(TAG) |
+              .metadata.annotations.containerImage = "quay.io/projectquay/quay-operator:"+env(TAG) |
+              .spec.version = strenv(TAG)+".999" |
+              del(.spec.replaces) |
+              .spec.install.spec.deployments[0].name = "quay-operator-test" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].image = "quay.io/projectquay/quay-operator:"+env(TAG) |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[4].value = "${{needs.quay-image.outputs.digest}}" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[5].value = "${{needs.clair-image.outputs.digest}}" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[6].value = "${{needs.builder-image.outputs.digest}}" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[7].value = "${{needs.qemu-builder-image.outputs.digest}}" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[8].value = "centos/postgresql-10-centos7@sha256:de1560cb35e5ec643e7b3a772ebaac8e3a7a2a8e8271d9e91ff023539b4dfb33" |
+              .spec.install.spec.deployments[0].spec.template.spec.containers[0].env[9].value = "centos/redis-32-centos7@sha256:06dbb609484330ec6be6090109f1fa16e936afcf975d1cbc5fff3e6c7cae7542"
+              ' bundle/downstream/manifests/quay-operator.clusterserviceversion.yaml
+
+      - name: Update Bundle Annotations
+        uses: mikefarah/yq@master
+        with:
+          cmd: |
+            yq eval -i '
+              .annotations."operators.operatorframework.io.bundle.channel.default.v1" = "test" |
+              .annotations."operators.operatorframework.io.bundle.channels.v1" = "test" |
+              .annotations."operators.operatorframework.io.bundle.package.v1" = "quay-operator-test"
+              ' bundle/downstream/metadata/annotations.yaml
+
+      - name: Publish Bundle Image
+        run: |
+          docker build -t "${BUNDLE}:${TAG}" -f ./bundle/Dockerfile ./bundle/downstream
+          docker push "${BUNDLE}:${TAG}"
```

Comment:
docker/build-push-action@v2

Reply:
I looked into using that action here too, but the docs states:

> Be careful because any file mutation in the steps that precede the build step will be ignored [...]

Thus we can't use it, since we want the changes on the previous step to be present in the image we're building.

Reply:
https://github.com/docker/build-push-action#usage

Reply:
Oh, interesting. I was not aware of that. I guess we could still use `context: .` though. Anyways, I am OK with whatever you decide.

Reply:
Ha! I guess I didn't finished reading the paragraph, nice catch. I'll try the context.

Reply:
tried this and couldn't get working on first try - I'll leave it as is for now just so I can move on from.